utils/ccdatofhir/mapping_resource_diagnostice_report.bal

```diff
@@ -0,0 +1,82 @@
+    if isXMLElementNotNull(organizerElement) {
+        r4:DiagnosticReport diagnosticReport = {code: {}, status: "partial"};
+
+        xml idElement = organizerElement/<v3:id|id>;
+        xml statusCodeElement = organizerElement/<v3:statusCode|statusCode>;
+        xml codeElement = organizerElement/<v3:code|code>;
+        xml translationElement = codeElement/<v3:translation|translation>;
+        xml effectiveTimeLowElement = organizerElement/<v3:effectiveTime|effectiveTime>/<v3:low|low>;
+        xml effectiveTimeHighElement = organizerElement/<v3:effectiveTime|effectiveTime>/<v3:high|high>;
+        xml effectiveTimeValueElement = organizerElement/<v3:effectiveTime|effectiveTime>/<v3:value|value>;
+
+        r4:Identifier[] identifier = [];
+        foreach xml idElem in idElement {
+            r4:Identifier? mapCcdaIdToFhirIdentifierResult = mapCcdaIdToFhirIdentifier(idElem);
+            if (mapCcdaIdToFhirIdentifierResult is r4:Identifier) {
+                identifier.push(mapCcdaIdToFhirIdentifierResult);
+            }
+        }
+        diagnosticReport.identifier = identifier;
+
+        diagnosticReport.status = mapCcdatoFhirDiagnosticReportStatus(statusCodeElement);
+
+        r4:CodeableConcept? mapCcdaCodingtoFhirCodeableConceptResult = mapCcdaCodingtoFhirCodeableConcept(translationElement);
+        if mapCcdaCodingtoFhirCodeableConceptResult is r4:CodeableConcept {
+            diagnosticReport.code = mapCcdaCodingtoFhirCodeableConceptResult;
+        } else {
+            mapCcdaCodingtoFhirCodeableConceptResult = mapCcdaCodingtoFhirCodeableConcept(codeElement);
+            if mapCcdaCodingtoFhirCodeableConceptResult is r4:CodeableConcept {
+                diagnosticReport.code = mapCcdaCodingtoFhirCodeableConceptResult;
+            }
+        }
+
+        r4:dateTime? mapCCDAEffectiveLowTimetoFHIRDateTimeResult = mapCcdaDateTimeToFhirDateTime(effectiveTimeLowElement);
+        if mapCCDAEffectiveLowTimetoFHIRDateTimeResult is r4:dateTime {
+            diagnosticReport.effectivePeriod.'start = mapCCDAEffectiveLowTimetoFHIRDateTimeResult;
+        }
+
+        r4:dateTime? mapCCDAEffectiveHighTimetoFHIRDateTimeResult = mapCcdaDateTimeToFhirDateTime(effectiveTimeHighElement);
+        if mapCCDAEffectiveHighTimetoFHIRDateTimeResult is r4:dateTime {
+            diagnosticReport.effectivePeriod.end = mapCCDAEffectiveHighTimetoFHIRDateTimeResult;
+        }
+
+        r4:dateTime? mapCCDAEffectiveValueTimetoFHIRDateTimeResult = mapCcdaDateTimeToFhirDateTime(effectiveTimeValueElement);
+        if mapCCDAEffectiveValueTimetoFHIRDateTimeResult is r4:dateTime {
+            diagnosticReport.effectiveDateTime = mapCCDAEffectiveValueTimetoFHIRDateTimeResult;
+        }
+        return diagnosticReport;
+    }
+    return ();
+}
+
+isolated function mapCcdatoFhirDiagnosticReportStatus(xml codingElement) returns r4:DiagnosticReportStatus {
+    string|error? codeVal = codingElement.data();
```

Comment:
```suggestion
    string codeVal = codingElement.data();
```
Check all the places

Reply:
Checked and updated in all other places in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/63d198b2a9b5526bb14c4cf256c08d18cea26095